node/actors/bft/src/testonly/ut_harness.rs

```diff
@@ -89,12 +119,23 @@ impl UTHarness {
         self.consensus.replica.view = view
     }
 
+    pub(crate) fn new_unfinalized_replica_prepare(&self) -> Signed<ConsensusMsg> {
+        self.new_current_replica_prepare(|msg| {
+            msg.high_qc = self.new_commit_qc(|high_qc| {
+                // Set view to the previous view since it needs to belong to an earlier view.
+                high_qc.view = msg.view.prev();
+                // Increment high_qc proposal number to create a discrepancy between the high_qc and the high_vote.
+                high_qc.proposal.number = high_qc.proposal.number.next();
```

Comment:
This is a very unnatural situation for a replica to be in. It must have voted for some block N on view M while not being aware of other blocks N and N+1 having been finalized on previous views. And it's impossible for a quorum of replicas to be in this situation.
It's preferable if the high vote here is for a view and block number greater than the high QC. Like this:

```suggestion
            let mut high_vote = ReplicaCommit {
                protocol_version: validator::ProtocolVersion::EARLIEST,
                view: self.consensus.replica.view.next(),
                proposal: self.consensus.replica.high_qc.message.proposal,
            };

            high_vote.proposal.parent = high_vote.proposal.hash();
            high_vote.proposal.number = high_vote.proposal.number.next();

            msg.high_vote = high_vote;
```